File: 3_claude_sol2.py
m = int(input())

temp = m
pf = []
i = 2
while i * i <= temp:
    if temp % i == 0:
        pf.append(i)
        while temp % i == 0:
            temp = temp // i
    i += 1
if temp > 1:
    pf.append(temp)

ans = 0
for x in range(1, m):
    for y in range(1, m):
        cursed = True
        for p in pf:
            if x % p != 0 and y % p != 0:
                cursed = False
                break
        if cursed:
            ans += 1

print(ans)

File: 3_claude_sol2.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m = sc.nextInt();

        int temp = m;
        ArrayList<Integer> pf = new ArrayList<>();
        for (int i = 2; i * i <= temp; i++) {
            if (temp % i == 0) {
                pf.add(i);
                while (temp % i == 0) {
                    temp /= i;
                }
            }
        }
        if (temp > 1) {
            pf.add(temp);
        }

        long ans = 0;
        for (int x = 1; x < m; x++) {
            for (int y = 1; y < m; y++) {
                boolean cursed = true;
                for (int i = 0; i < pf.size(); i++) {
                    int p = pf.get(i);
                    if (x % p != 0 && y % p != 0) {
                        cursed = false;
                        break;
                    }
                }
                if (cursed) {
                    ans++;
                }
            }
        }

        System.out.println(ans);
    }
}
